packages/rich-text/src/to-tree.js

```diff
@@ -34,7 +35,7 @@ export function toTree( value, multilineTag, settings ) {
 		onStartIndex,
 		onEndIndex,
 	} = settings;
-	const { formats, text, start, end } = value;
+	const { formats, text, start, end } = normaliseFormats( value );
```

Comment:
I know `normaliseFormats` is performant at the moment but this is the kind of function that can hurt the editor's performance pretty quickly if we made a not-good-enough change. How can we ensure it doesn't happen? Any ideas.

(don't require an immediate answer)